src/main/java/teho/main/TehO.java

```diff
@@ -0,0 +1,199 @@
+package teho.main;
+
+import teho.exceptions.*;
+
+import java.util.Scanner;
+import static java.lang.Integer.parseInt;
+import java.time.LocalDate;
+
+/**
+ * Main class that represents a Personal Assistant Chatbot
+ * that helps a person to keep track of various things.
+ */
+public class TehO  {
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    /**
+     * Constructs a TehO instance with string representation of a file path
+     * to save and load tasks into and from.
+     *
+     * @param filePath Path of file to save and load tasks into and from.
+     */
+    public TehO(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        taskList = new TaskList(storage.loadTasks());
+    }
+
+    public String getResponse(String userCommand) {
+        String returnMessage = null;
+            if (userCommand.equals("bye")) {
+                storage.saveTasks(taskList);
+                returnMessage = ui.generateGoodbyeMessage();
+            } else if (userCommand.equals("list")) {
+                returnMessage = ui.generateList(taskList);
+            } else if (userCommand.startsWith("mark")) {
+                markTask(userCommand);
+                returnMessage = ui.generateUnmarkTaskMessage(taskList.getTask(parseInt(userCommand.split(" ")[1]) - 1));;
```

Comment:
Not sure I like how complicated the returnMessage expressions are, especially with multiple layers of nested parenthesis.  It might be better for readability if it was done in steps.
```suggestion
int index = parseInt(userCommand.split(" ")[1]) - 1));
Task taskUnmarked = taskList.getTask(index);
returnMessage = ui.generateUnmarkTaskMessage(taskUnmarked);
```
(Something like this maybe? Assuming I managed to understand the code correctly...)